classes/local/store/s3/client.php

```diff
@@ -571,26 +571,32 @@ public function get_expiration_time($now, $expires) {
             // Convert to a valid timestamp.
             $expires = strtotime($expires);
         }
+        // Invalid or already expired:
         // If it's set to 0 or strtotime() returned false,
         // set it to default + 1 min as a healthy margin.
         if (empty($expires)) {
             $expires = $now + $this->expirationtime + MINSECS;
         }
+        // Expiry too short, push it out to the next 2 minutes (will round down later):
         // If it's already expired or expires less than in a minute,
-        // set it to 1 minute.
+        // set it to 2 minutes. This works together with rounding down later on
+        // to ensure a non zero expiry time, and a minimum expiry of 1 minute.
         if ($expires < $now + MINSECS) {
-            $expires = $now + MINSECS;
+            $expires = $now + (2 * MINSECS);
         }
+        // Checks upper bound:
         // The expiration date of a signature version 4 presigned URL must be
         // less than one week. So if it's greater than a week set it to 1 week.
         // Use MINSECS as a healthy margin of error.
         if ($expires - $now > WEEKSECS - MINSECS) {
             $expires = $now + WEEKSECS - MINSECS;
         }
-        if (is_null($expires) || false === $expires) {
```

Comment:
This block would never have been reached, and is already covered by checks in lines 576-578 (prev commit)